src/components/furniture/Separators.vue

```diff
@@ -93,46 +94,58 @@ export default {
         height: this.shelf.height * 3 + 'px',
       };
     },
-    shelfIndex() {
-      return this.shelfs.findIndex((shelf) => shelf.id === this.shelf.id);
-    },
-    shelfSeparators() {
-      return this.shelfs[this.shelfIndex].insideSeparators;
-    },
     widthVariation() {
-      if (this.typeOfSeparators === 'centered') {
+      if (this.separators.typeOfSeparators === 'centered') {
         return 0;
-      } else if (this.typeOfSeparators === 'left') {
-        return -12 / this.amountOfSeparators;
+      } else if (this.separators.typeOfSeparators === 'left') {
+        return -12 / this.separators.amountOfSeparators;
       } else {
-        return 12 / this.amountOfSeparators;
+        return 12 / this.separators.amountOfSeparators;
       }
     },
     separatorWidth() {
-      if (this.amountOfSeparators === '1') {
+      if (this.separators.amountOfSeparators === '1') {
         return 50 + this.widthVariation;
-      } else if (this.amountOfSeparators === '2') {
+      } else if (this.separators.amountOfSeparators === '2') {
         return 33.3 + this.widthVariation;
-      } else if (this.amountOfSeparators === '3') {
+      } else if (this.separators.amountOfSeparators === '3') {
         return 25 + this.widthVariation;
       } else {
         return 100;
```

Comment:
Usually, long if/else are a "code smell". It means that it's not soooo baaad but that there is a way cooler way to do it.

So what does this function? It checks a number (let's assume here you fixed the string issue and that this.separators.typeOfSeparators can be the number 1, 2 or 3).

And for each number, it returns a special value. 

So that is a mapping basically. Each value is mapped to another value.

1 => 50
2 => 33.3
3 => 25
undefined => 100

Right? 

We have several tools that allow us to do this kind of mapping. 

An array is a good example. array[x] => y  

Or an object too! object[x] => y

Both arrays and objects have limitations. 

But Javascript introduced recently Map which is exactly what we need now and that allow us to associate anything to anything. 

Let's try to refactor:

This: 
```javascript 
    separatorWidth() {
      if (this.separators.amountOfSeparators === 1) {
        return 50 + this.widthVariation;
      } else if (this.separators.amountOfSeparators === 2) {
        return 33.3 + this.widthVariation;
      } else if (this.separators.amountOfSeparators === 3 {
        return 25 + this.widthVariation;
      } else {
        return 100;
     }
   }
```

Could become: 

```javascript 
const mapSeparatorWidth = new Map([
  [1, 50],
  [2, 33.3],
  [3, 25],
  ['default', 100]
]);

// ... and below 

    separatorWidth() {
      return mapSeparatorWidth.get(this.separators.amountOfSeparators) || mapSeparatorWidth.get('default');
   }

```

That's it! 

So if `this.separators.amountOfSeparators` is some weird shit like `52`, the `.get(52)` will returned `undefined` and then the other side of the `||` operator will apply, getting the default value.